### ISSUE TITLE
Queued cranelift-fuzzgen enhancements

### ISSUE THREAD

**Issue by jameysharp** [bytecodealliance/wasmtime#4798]
We got a large batch of new fuzz-bugs from cranelift-fuzzgen (and cranelift-icache) this week. Some of them are bugs we don't want to fix yet because they're just x64 lowerings that aren't implemented yet, or things that we do want to test eventually but haven't built all the testing infrastructure for yet. We're trying to make sure that all the bugs we _do_ want to fix are fully dealt with before we start changing fuzzing input formats and invalidating our current test cases.

Check off these issues once they are addressed by either an already-merged fix, or by a pending PR that we aren't ready to merge yet:

- [x] #4755
  - fixed by #4752
- [x] #4756
  - fixed by #4771
- [x] #4757
  - fixed by #4797
- [x] #4758
  - fixed by #4797
- [x] #4759
  - fixed by #4768
- [x] #4760
  - fixed by #4752
- [x] #4761
  - fixed by #4790
- [x] #4770
  - fixed by #4771

And these are the PRs that we need to merge when we're satisfied with our investigations of the above issues:

- #4771
- #4782
- #4783 (deferred from #4752)
- #4797

I think we're ready to merge these. @afonso360 and @cfallin, do you agree?

**Comment by afonso360** [comment]
Yeah, I think we should be good to merge those. I'm really happy that fuzzgen is finding real bugs like https://github.com/bytecodealliance/wasmtime/pull/4752 and https://github.com/bytecodealliance/wasmtime/pull/4790 🎉

**Comment by jameysharp** [comment]
...I guess I should close this since all the PRs I was tracking are merged now.
